submissions/Valkirin/frontend-nanodegree-arcade-game/js/app.js

```diff
@@ -0,0 +1,80 @@
+var Enemy = function(x, y, speed) {
+    this.x = x;
+	this.y = y;
+	this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function(dt) {
+    this.x +=  + (this.speed * dt);
+    
+    if (this.x > 550) {
+    	this.x = -100;
+		this.speed = Math.floor(100 + (Math.random() * 200));
+    };
+    
+    if (player.x > this.x - 60 && 
+        player.x < this.x + 60 && 
+        player.y > this.y - 60 &&
+        player.y < this.y + 60) {
+        player.x = 200;
+        player.y = 400;
+	};
+    
+};
+
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+var allEnemies = []
+var enemyLocation = [60, 140, 220]
+
+enemyLocation.forEach(function (y) {
+    allEnemies.push(new Enemy(0, y, 300))
+});
```

Comment:
Perfectly chainable. Here is the hint:

`var objectsContainer = [1,2,3].map(x => new Object(x));` or 
`var objectsContainer = semanticallyNamedArrayContainer.map(x => new Object(x));`

Reply:
fixed